### PR TITLE
Feature/apl 1418 deleted column

### PR DESCRIPTION
**Pull request by AndrewBoyarsky**

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/derived/BasicDbTable.java [hunk 1]
```diff
@@ -248,6 +250,32 @@ private int deleteDeletedNewAlgo(int height) throws SQLException {
         return deleted;
     }
 
+    private int deleteDeletedBasedOnDeletedColumn(int height) throws SQLException {
```
Comment (alzinchenko) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/derived/BasicDbTable.java]: What about more clear method name

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/derived/BasicDbTable.java [hunk 2]
```diff
@@ -60,7 +63,7 @@ private int doMultiversionRollback(int height) {
         }
         long startTime = System.currentTimeMillis();
         String sql = "UPDATE " + table
-                + " SET latest = TRUE " + keyFactory.getPKClause() + " AND height ="
+                + " SET latest = TRUE " + (supportDelete() ? ", deleted = FALSE " : "") + keyFactory.getPKClause() + " AND height ="
```
Comment (silaev) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/derived/BasicDbTable.java]: What about extract '(supportDelete() ? ", deleted = FALSE " : "")' into a method called like addtDeleteIfPresent to ease reading this query?

Comment (silaev-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/derived/BasicDbTable.java]: What about extract '(supportDelete() ? ", deleted = FALSE " : "")' into a method called like addtDeleteIfPresent to ease reading this query?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/model/VersionedDeletableEntity.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import lombok.ToString;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+public class VersionedDeletableEntity extends VersionedDerivedEntity {
+    @Getter
+    @Setter
+    private boolean deleted = false;
```
Comment (silaev-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/model/VersionedDeletableEntity.java]: it's false by default (see https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html, Default Values)

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/derived/BasicDbTable.java (hunk 2 above)
Comment (AndrewBoyarsky) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/derived/BasicDbTable.java]: done

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/derived/BasicDbTable.java (hunk 1 above)
Comment (AndrewBoyarsky) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/derived/BasicDbTable.java]: Method was deleted completely, another implementation used instead.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/model/VersionedDeletableEntity.java (hunk 3 above)
Comment (AndrewBoyarsky) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/model/VersionedDeletableEntity.java]: done

**Pull request merged**